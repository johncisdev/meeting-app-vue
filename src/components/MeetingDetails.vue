<template>
  <v-container class="grey darken-4" fluid style="height: calc(100vh - 50px)">
    <v-row>
      <v-col
        cols="12"
        sm="8"
        class="d-flex justify-center align-center"
        style="height: calc(100vh - 100px)"
      >
        <v-avatar
          class="white--text"
          color="primary"
          size="128"
          v-text="hostName"
          style="font-size: 3rem"
        ></v-avatar
      ></v-col>
      <v-col cols="12" sm="4" class="d-flex justify-center align-center"
        ><v-card elevation="2">
          <v-card-text>
            <div class="text-h6 mb-6">Meeting Details</div>
            <v-row justify="center">
              <v-expansion-panels v-model="panel" multiple>
                <v-expansion-panel key="0">
                  <v-expansion-panel-header hide-actions ripple
                    >Info</v-expansion-panel-header
                  >
                  <v-expansion-panel-content>
                    <v-row>
                      <v-col cols="12" sm="5">
                        <v-card-subtitle class="pa-0 text--secondary">
                          Room Name
                        </v-card-subtitle>
                        <v-card-text class="text--primary pb-0">
                          <div id="roomName">{{ meeting.name }}</div>
                        </v-card-text>
                      </v-col>
                      <v-col cols="12" sm="7">
                        <v-card-subtitle class="pa-0 text--secondary">
                          Host
                        </v-card-subtitle>
                        <v-card-text class="text--primary pb-0">
                          <div id="hostName">{{ meeting.hostName }}</div>
                        </v-card-text>
                      </v-col>
                    </v-row>

                    <v-row>
                      <v-col cols="12" sm="5">
                        <v-card-subtitle class="pa-0 text--secondary">
                          Date
                        </v-card-subtitle>
                        <v-card-text class="text--primary pb-0">
                          <div id="bookingDate">{{ meeting.bookingDate }}</div>
                        </v-card-text>
                      </v-col>
                      <v-col cols="12" sm="7">
                        <v-card-subtitle class="pa-0 text--secondary">
                          Time
                        </v-card-subtitle>
                        <v-card-text class="text--primary pb-0">
                          <div>
                            <span id="bookingStartTime">{{
                              meeting.bookingStartTime
                            }}</span>
                            -
                            <span id="bookingEndTime">
                              {{ meeting.bookingEndTime }}</span
                            >
                          </div>
                        </v-card-text>
                      </v-col>
                    </v-row>
                  </v-expansion-panel-content>
                </v-expansion-panel>
                <v-expansion-panel key="1">
                  <v-expansion-panel-header hide-actions ripple
                    >Participants</v-expansion-panel-header
                  >
                  <v-expansion-panel-content>
                    <v-list subheader>
                      <v-subheader>In the meeting</v-subheader>

                      <v-list-item
                        v-for="participant in meeting.participants"
                        :key="participant"
                      >
                        <v-list-item-avatar>
                          <v-avatar
                            class="accent white--text mr-1"
                            size="32"
                            v-text="participant.slice(0, 1).toUpperCase()"
                          >
                          </v-avatar>
                        </v-list-item-avatar>

                        <v-list-item-content>
                          <v-list-item-title
                            v-text="participant"
                          ></v-list-item-title>
                        </v-list-item-content>
                      </v-list-item>
                    </v-list>
                  </v-expansion-panel-content>
                </v-expansion-panel>
              </v-expansion-panels>
            </v-row>
            <v-row>
              <v-col cols="12">
                <v-btn color="error" block @click="$router.push({ path: '/' })">
                  Leave meeting
                </v-btn>
              </v-col>
            </v-row>
          </v-card-text>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  name: "Meeting",
  data() {
    return {
      meeting: {},
      hostName: "",
      panel: [0, 1],
    };
  },
  async created() {
    await this.$store.dispatch("getMeeting", this.$route.params.meetingId);
    this.meeting = this.$store.state.data[0];
    const host = this.$store.state.data[0].hostName.split(" ");
    this.hostName =
      host.length &&
      `${host[0].slice(0, 1).toUpperCase()}${
        host[1] ? host[1].slice(0, 1).toUpperCase() : ""
      }`;
  },
};
</script>
