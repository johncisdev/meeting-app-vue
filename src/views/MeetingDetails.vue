<template>
  <meeting-details />
</template>

<script>
import MeetingDetails from "../components/MeetingDetails";

export default {
  name: "Meeting",

  components: {
    MeetingDetails,
  },
};
</script>
