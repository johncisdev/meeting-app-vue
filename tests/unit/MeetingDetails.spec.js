import { shallowMount, createLocalVue, mount } from "@vue/test-utils";
import Vuex from "vuex";

import MeetingDetails from "@/components/MeetingDetails.vue";
import Store from "@/store/index";

describe("MeetingDetails.vue test suite", () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let store;
  let propsData;
  const DATA = {
    name: "Room 101",
    hostName: "Chris",
    participants: ["Nang", "Let", "Sal", "Ming"],
    bookingDate: "11/11/2021",
    bookingStartTime: "10:00",
    bookingEndTime: "10:30",
    id: 4,
  };

  beforeEach(() => {
    store = new Vuex.Store(Store);
    propsData = {};
  });

  it("should match with the snapshot", () => {
    const wrapper = mount(MeetingDetails, {
      store,
      propsData,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should render the component", () => {
    const wrapper = shallowMount(MeetingDetails, {
      store,
      propsData,
    });
    expect(wrapper.exists()).toBe(true);
  });

  it("should render room and meeting info", async () => {
    const $store = {
      state: { data: DATA },
    };
    const wrapper = shallowMount(MeetingDetails, {
      propsData,
      mocks: {
        $store,
        $route: {
          params: { meetingId: 4 },
        },
      },
    });

    await wrapper.setData({
      meeting: DATA,
    });

    //check if rendered
    expect(wrapper.find("#roomName").text()).toBe(DATA.name);
    //check in store
    expect(wrapper.vm.$store.state.data.name).toBe(DATA.name);

    //check if rendered
    expect(wrapper.find("#hostName").text()).toBe(DATA.hostName);
    //check in store
    expect(wrapper.vm.$store.state.data.hostName).toBe(DATA.hostName);

    //check if rendered
    expect(wrapper.find("#bookingDate").text()).toBe(DATA.bookingDate);
    //check in store
    expect(wrapper.vm.$store.state.data.bookingDate).toBe(DATA.bookingDate);

    //check if rendered
    expect(wrapper.find("#bookingStartTime").text()).toBe(
      DATA.bookingStartTime
    );
    //check in store
    expect(wrapper.vm.$store.state.data.bookingStartTime).toBe(
      DATA.bookingStartTime
    );

    //check if rendered
    expect(wrapper.find("#bookingEndTime").text()).toBe(DATA.bookingEndTime);
    //check in store
    expect(wrapper.vm.$store.state.data.bookingEndTime).toBe(
      DATA.bookingEndTime
    );
  });
});
