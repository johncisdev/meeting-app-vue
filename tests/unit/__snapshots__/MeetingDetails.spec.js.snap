// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingDetails.vue test suite should match with the snapshot 1`] = `
<v-container
  class="grey darken-4"
  fluid=""
  style="height: calc(100vh - 50px);"
>
  <v-row>
    <v-col
      class="d-flex justify-center align-center"
      cols="12"
      sm="8"
      style="height: calc(100vh - 100px);"
    >
      <v-avatar
        class="white--text"
        color="primary"
        size="128"
        style="font-size: 3rem;"
      />
    </v-col>
     
    <v-col
      class="d-flex justify-center align-center"
      cols="12"
      sm="4"
    >
      <v-card
        elevation="2"
      >
        <v-card-text>
          <div
            class="text-h6 mb-6"
          >
            Meeting Details
          </div>
           
          <v-row
            justify="center"
          >
            <v-expansion-panels
              multiple=""
            >
              <v-expansion-panel>
                <v-expansion-panel-header
                  hide-actions=""
                  ripple=""
                >
                  Info
                </v-expansion-panel-header>
                 
                <v-expansion-panel-content>
                  <v-row>
                    <v-col
                      cols="12"
                      sm="5"
                    >
                      <v-card-subtitle
                        class="pa-0 text--secondary"
                      >
                        
                        Room Name
                      
                      </v-card-subtitle>
                       
                      <v-card-text
                        class="text--primary pb-0"
                      >
                        <div
                          id="roomName"
                        >
                          
                        </div>
                      </v-card-text>
                    </v-col>
                     
                    <v-col
                      cols="12"
                      sm="7"
                    >
                      <v-card-subtitle
                        class="pa-0 text--secondary"
                      >
                        
                        Host
                      
                      </v-card-subtitle>
                       
                      <v-card-text
                        class="text--primary pb-0"
                      >
                        <div
                          id="hostName"
                        >
                          
                        </div>
                      </v-card-text>
                    </v-col>
                  </v-row>
                   
                  <v-row>
                    <v-col
                      cols="12"
                      sm="5"
                    >
                      <v-card-subtitle
                        class="pa-0 text--secondary"
                      >
                        
                        Date
                      
                      </v-card-subtitle>
                       
                      <v-card-text
                        class="text--primary pb-0"
                      >
                        <div
                          id="bookingDate"
                        >
                          
                        </div>
                      </v-card-text>
                    </v-col>
                     
                    <v-col
                      cols="12"
                      sm="7"
                    >
                      <v-card-subtitle
                        class="pa-0 text--secondary"
                      >
                        
                        Time
                      
                      </v-card-subtitle>
                       
                      <v-card-text
                        class="text--primary pb-0"
                      >
                        <div>
                          <span
                            id="bookingStartTime"
                          >
                            
                          </span>
                          
                          -
                          
                          <span
                            id="bookingEndTime"
                          >
                            
                            
                          </span>
                        </div>
                      </v-card-text>
                    </v-col>
                  </v-row>
                </v-expansion-panel-content>
              </v-expansion-panel>
               
              <v-expansion-panel>
                <v-expansion-panel-header
                  hide-actions=""
                  ripple=""
                >
                  Participants
                </v-expansion-panel-header>
                 
                <v-expansion-panel-content>
                  <v-list
                    subheader=""
                  >
                    <v-subheader>
                      In the meeting
                    </v-subheader>
                     
                  </v-list>
                </v-expansion-panel-content>
              </v-expansion-panel>
            </v-expansion-panels>
          </v-row>
           
          <v-row>
            <v-col
              cols="12"
            >
              <v-btn
                block=""
                color="error"
              >
                
                Leave meeting
              
              </v-btn>
            </v-col>
          </v-row>
        </v-card-text>
      </v-card>
    </v-col>
  </v-row>
</v-container>
`;
